### PR TITLE
Add support for no button to `StatefulBrowser.submit_selected`

### LINKED ISSUE #403: Submit form with no button
Some web pages use JavaScript postbacks to submit forms. This means that a submitted form has nothing to do with which button is pressed. In fact, including form data for the "pressed" button can mess things up in such a case.

Accordingly, it would be great to have a way of specifying that *no* button "press" should be simulated when calling `submit_selected`. Right now, `btnName = None` means that the first "submit" button is auto-selected — but there is no way to specify that *no* submit button press should be simulated. Perhaps another (boolean) parameter could be used for this? The `Form.choose_submit` method would also need to be enhanced accordingly.

### PR DESCRIPTION
**Pull request by alexreg**
Resolves #403

### CODE REVIEW

File: tests/test_stateful_browser.py
Comment (hemberger) [inline tests/test_stateful_browser.py:190]: We may need to rethink the implementation of this change, because it posts _all_ of the submit elements, rather than none of them:

```html
        <input type="submit" name="preview" value="Preview Page" />
        <input type="submit" name="diff" value="Review Changes" />
        <input type="submit" id="save" name="save" value="Submit changes" />
        <button type="submit" name="cancel" value="Cancel" />
```
I assume this is not the intended behavior? If we don't want any of the submit elements to be selected, then we probably need to modify `form.choose_submit` to delete all submit inputs if `submit is False` (or perhaps a new method with some refactoring).

**Review by hemberger — CHANGES_REQUESTED** [review]
Thanks for the PR! This looks like what we discussed in #403, except it seems to contain some unrelated changes. Maybe you pushed some commits you did not intend to push?

File: mechanicalsoup/stateful_browser.py
Comment (alexreg) [reply mechanicalsoup/stateful_browser.py:174]: This was an accident, sorry.

File: tests/test_stateful_browser.py
Comment (alexreg) [reply tests/test_stateful_browser.py:190]: Yeah, I was wondering about that... Wasn't sure what the behaviour ought to be, but that's fair enough. I'll have a look into it.

**Comment by alexreg** [comment]
@hemberger Thanks for having a look. I've addressed your comments, and the current solution may more along the right lines.

File: tests/test_stateful_browser.py
Comment (hemberger) [inline tests/test_stateful_browser.py:186]:
```suggestion
    '''Tests that no submit inputs are posted when btnName=False.'''
```

File: mechanicalsoup/form.py
Comment (hemberger) [inline mechanicalsoup/form.py:313]: Would you mind extending the documentation of the `False` case to the `btnName` argument of `stateful_browser.submit_selected`?

**Review by hemberger — CHANGES_REQUESTED** [review]
This looks like it works as expected now, thanks for the fixes! I have two very minor suggestions, but once those are addressed I will be happy to approve this PR. Thanks again for the submission!

File: mechanicalsoup/form.py
Comment (alexreg) [reply mechanicalsoup/form.py:313]: Sure. I had actually left it in initially, then removed it for economy's sake, but it's probably helpful to have it in there, indeed.

**Comment by alexreg** [comment]
@hemberger I've now expanded the docs for that method (and even reworked it slightly). Hope that's alright.

**Review by hemberger — APPROVED** [review]
LGTM, thanks again for your contribution! If there are no other comments from other maintainers, I'll merge this in a day or so.

**Pull request merged**
